test/gtest/ucs/test_event_set.cc

```diff
@@ -0,0 +1,141 @@
+/**
+* Copyright (C) Hiroyuki Sato. 2019. ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#include <common/test.h>
+extern "C" {
+#include <ucs/sys/event_set.h>
+#include <sys/epoll.h>
+}
+
+class test_event_set : public ucs::test {
+};
+
+static const char* const UCS_EVENT_SET_TEST_STRING = "OpenUCX event_set test";
+
+UCS_TEST_F(test_event_set, event_set_create) {
+    ucs_sys_event_set_t *event_set = NULL;
+    ucs_status_t status;
+
+    status = ucs_event_set_create(&event_set);
+    EXPECT_EQ(UCS_OK,status);
+    EXPECT_TRUE(event_set != NULL);
+
+    ucs_event_set_cleanup(event_set);
+}
+
+#if 0
+/*
+ * [1554806179.102506] [gorilla:30978:0]      event_set.c:102  \
+ *   UCX  ERROR epoll_ctl(epfd=3, ADD, fd=-1) failed: Bad file descriptor
+ * ucs/test_event_set.cc:39: Failure
+ * Value of: status
+ *   Actual: -3
+ * Expected: UCS_ERR_INVALID_PARAM
+ * Which is: -5
+ * common/test.cc:251: Failure
+ * Failed
+ * Got 1 errors during the test
+ * [  FAILED  ] test_event_set.event_set_add, where TypeParam =  and \
+ *  GetParam() =  (0 ms)
+ * [----------] 2 tests from test_event_set (0 ms total)
+ * 
+ * [----------] Global test environment tear-down
+ * [==========] 2 tests from 1 test case ran. (0 ms total)
+ * [  PASSED  ] 1 test.
+ * [  FAILED  ] 1 test, listed below:
+ * [  FAILED  ] test_event_set.event_set_add, where TypeParam =  and \
+ *  GetParam() =
+ * 
+ *  1 FAILED TEST
+ */
+UCS_TEST_F(test_event_set, event_set_add) {
+    ucs_sys_event_set_t *event_set = NULL;
+    ucs_status_t status;
+
+    status = ucs_event_set_create(&event_set);
+    EXPECT_EQ(UCS_OK,status);
+    EXPECT_TRUE(event_set != NULL);
+
+    status = ucs_event_set_add(event_set,-1,UCS_EVENT_SET_EVREAD,NULL);
+    EXPECT_EQ(UCS_ERR_INVALID_PARAM,status);
+    ucs_event_set_cleanup(event_set);
+}
+#endif
+
+#if 0
+/*
+ * ucs_event_set_events_map is the static function.
+ * Disable until testable.
+ *
+ */
+UCS_TEST_F(test_event_set, ucs_event_set_events_map) {
+    int ret;
```

Comment:
Do you know is it possible to test `static` function?

Reply:
If it is static inline it is possible, otherwise static is like private for the C file.

Reply:
Thanks, @shamisp 
It is impossible to test `static inline xxx function` using gtest. Is this correct?
If correct, I'll change like the following and remove this test.

```diff
diff --git a/src/ucs/sys/event_set.c b/src/ucs/sys/event_set.c
index d173f659..1611d196 100644
--- a/src/ucs/sys/event_set.c
+++ b/src/ucs/sys/event_set.c
@@ -28,7 +28,7 @@ struct ucs_event_set_type_table {
 };

 /* int ucs_event_set_events_map(int events) */
-static int ucs_event_set_events_map(int events)
+static inline int ucs_event_set_events_map(int events)
```

Reply:
Is the function is used anywhere outside of event_set.c file ? From brief look at appears that this is local function for that file. If this is the case it can be declared as static inline. No need to do gtest for this. We gtest mostly interface functions.

Reply:
Thanks @shamisp . I removed this test.